Gemfile management

deivid-rodriguez opened the pull request:
This is my proposal after the discussion in #453.

Closes #445.

deivid-rodriguez committed:
Move latest Gemfile to repo root

So that contributors have an initial working bundle.

deivid-rodriguez committed:
Add lockfiles to the repo

So the resulting environment for a given Gemfile is deterministic.

deivid-rodriguez committed:
Test against latest dependencies in CI

This is not very good for contributors, since they might get a failing
build when it's not their fault and build passes locally, but it's good
for discovering incompatibilities in latest dependencies.

We decided to go for the latter.

deivid-rodriguez committed:
Remove deployment bundler arg

deivid-rodriguez commented:
I should say that I'm not a fan of the `rm ${BUNDLE_GEMFILE}.lock` trick but I wanted to find the middle ground, so here it is!

rafaelfranca commented on .travis.yml:
```diff
@@ -5,11 +5,16 @@ rvm:
+  - Gemfile
+
+script:
+  - rm "${BUNDLE_GEMFILE}.lock" && bundle exec rake
```
I think the rm should be before bundle runs. So a `before_install` hook like this one? https://github.com/rails/rails/blob/master/.travis.yml#L34

deivid-rodriguez replied:
Oooops, you're right! I'm surprised this solution didn't actually fail!

deivid-rodriguez committed:
Move lock file removal to before_install

So that it's run before bundle install. We can also remove back the
default `script:` as it was before.

deivid-rodriguez committed:
Revert "Remove deployment bundler arg"

This reverts commit ea2e085f7b5b9d8f5fa9add33f7d6be402eee6e9.

I think the reason I needed this was that my initial patch was
incorrect.

deivid-rodriguez commented:
It should be fixed now!

javierjulio commented:
@deivid-rodriguez thanks for doing this for all the repos! And thanks to you @rafaelfranca for helping us straighten this out. Much appreciated! 🙏🏻❤️
